components/lsp-form/LongShort.tsx

```diff
@@ -35,6 +36,56 @@ const LongShort: FC<Props> = ({
   shortTokenBalance,
   shortTokenDecimals,
 }) => {
+  const setCollateralInput = useCallback(
+    (
+      tokenAmount: string,
+      valueCallback: React.Dispatch<React.SetStateAction<string>>
+    ) => {
+      const normalizedCPP = ethers.utils.formatEther(collateralPerPair);
+
+      const newAmount = Number(tokenAmount) * Number(normalizedCPP);
+
+      setAmount(newAmount.toString());
+      valueCallback(tokenAmount);
+    },
+    [collateralPerPair]
+  );
+
+  const maxTokensRedeemable = useCallback(() => {
+    const normalizedCPP = ethers.utils.formatEther(collateralPerPair);
+
+    let ltb = "0",
+      stb = "0",
+      am = "0";
+
+    if (longTokenBalance.gte(shortTokenBalance)) {
+      ltb = ethers.utils.formatUnits(shortTokenBalance, shortTokenDecimals);
+      stb = ethers.utils.formatUnits(shortTokenBalance, shortTokenDecimals);
+      am = (
+        Number(
+          ethers.utils.formatUnits(shortTokenBalance, shortTokenDecimals)
+        ) * Number(normalizedCPP)
+      ).toString();
+    }
+
+    if (longTokenBalance.lt(shortTokenBalance)) {
+      ltb = ethers.utils.formatUnits(longTokenBalance, longTokenDecimals);
+      stb = ethers.utils.formatUnits(longTokenBalance, longTokenDecimals);
+      am = (
+        Number(
+          ethers.utils.formatUnits(
+            Number(longTokenBalance.toString()) * Number(normalizedCPP),
+            longTokenDecimals
+          )
+        ) * Number(normalizedCPP)
+      ).toString();
+    }
+
+    setShortTokenAmount(stb);
+    setLongTokenAmount(ltb);
+    setAmount(am);
```

Comment:
just a nit here but theres a lot of duplicate lines since stb and ltb are always equal. am is just a variation of this, so u could prob simplify to just setting a single var in the branches above, essentially the smaller of the 2 values, then setting these 3 based on that single val

Reply:
Good point.